src/ng/parse.js

```diff
@@ -102,6 +102,16 @@ function ensureSafeFunction(obj, fullExpression) {
   }
 }
 
+function ensureSafeAssignContext(obj, fullExpression) {
+  if (obj) {
+    if (obj === (0).constructor || obj === (false).constructor || obj === ''.constructor ||
+        obj === {}.constructor || obj === [].constructor || obj === Function.constructor) {
+      throw $parseMinErr('isecaf',
+        'Assigning to a constructor is disallowed! Expression: {0}', fullExpression);
+    }
+  }
```

Comment:
we care the ones an expression can invoke and can be used to break the sandbox... the only other I can think about is `Scope`, but I really have no idea if it can be maliciously used. The alternative would be blocking to any property named `constructor`, but it might be just too much and there might be valid uses.